[DEVOPS-365] Slight (slight) improvement to Dockerfile.

jmfadeley opened the pull request:
**What's this PR do?**
Slightly tweaks the process to allow for correct deployments in Docker. Unfortunately, the survey microservice does not really build a production ready deployment right now. I've created [DEVOPS-366] to handle that. 

**Related JIRA tickets:**
DEVOPS-365, DEVOPS-366.

**How should this be manually tested?**
First, create a network if you haven't already:
`docker network create indaba-network`

Ensure that the docker instance has a database to work with. We're assuming the username and password in your .env are both set to amida, if not then please change. To start a docker instance of a database, run:
`docker run -p 5432:5432 --name survey-db --network indaba-network -e POSTGRES_USER=amida -e POSTGRES_PASSWORD=amida -e POSTGRES_DB=amida_survey_microservice postgres:9.6`

Then run a build in Docker with:
`docker build -t amida-survey-microservice .`

Finally try a deployment:
`docker run -p 9005:9005 --network indaba-network --name amida-survey-microservice -v <ABSOLUTE PATH>/.env:/app/.env -e SURVEY_SERVICE_PG_HOST=survey-db -e SURVEY_SERVICE_PG_USER=amida -e SURVEY_SERVICE_PG_PASSWORD=amida amida-survey-microservice yarn seed && yarn serve`

If you have any Postman requests, fire them at the endpoint to confirm completion.

**Any background context you want to provide?**
This blocks an ease-of-deployment for Indaba. I am currently trying to get four instances running for four people as simply as possible to cut down on... hour drain.

**Screenshots (if appropriate):**

jmfadeley committed:
[DEVOPS-365] Slight tweak to Dockerfile. However, it doesn't build a
production deployment. I've created DEVOPS-366 to handle that.

jmfadeley committed:
[DEVOPS-365] Adding .dockerignore.

jmfadeley committed:
Backup of changes.

jmfadeley committed:
[DEVOPS-365] Staged a builder and modified the package for deployments.

jmfadeley committed:
Tweak of .env.example.

jmfadeley committed:
[DEVOPS-365] Forgot seed in package.json.

jmfadeley committed:
[DEVOPS-365] Tweak.

jmfadeley committed:
[DEVOPS-365] Changes to give underscores to DB example.

mountHouli reviewed (approved):
The code looks good to me.

I didn't test anything because I have never run the survey service before.

rmharrison commented on Dockerfile:
```diff
@@ -1,18 +1,20 @@
+COPY . /app/
+
+# Run yarn
+RUN yarn install --pure-lockfile
```
You want `--production --frozen-lockfile`, presumably?

rmharrison replied:
Nope. Non-production is intentional.

Proximal
- cross-env is in dev deps, and is required by `create_db`, `delete_db` and `seed`

Root
- `yarn serve` requires some dev deps because there's not build step
- config isn't entirely loaded from `.env`
- seed script isn't on sequelize.

rmharrison reviewed (commented):
```
$node seed.js

...
{ SequelizeConnectionError: database "surveyServicetest" does not exist
```

Does this repo need the `create_db` and `delete_db` scripts as well?

rmharrison commented on test/shared.integration.js:
```diff
@@ -13,7 +13,7 @@ const SharedIntegration = require('./util/shared-integration.js');
 
 const models = require('../models');
-const config = require('../config');
+const config = require('../config/config');
```
```
$ yarn test

Error: Cannot find module '../config/config'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:580:15)
    at Function.Module._load (internal/modules/cjs/loader.js:506:25)
    at Module.require (internal/modules/cjs/loader.js:636:17)
    at require (internal/modules/cjs/helpers.js:20:18)
    at Object.<anonymous> (/home/work/git/amida-tech/amida-survey-microservice/test/shared.integration.js:16:16)
```

rmharrison replied:
Test (5464) pass for both `const config = require('../config');` and `const config = require('../config/test');`

rmharrison commented on CHANGELOG.md:
```diff
@@ -0,0 +1,5 @@
+# Changelog
```
Please remove windows end of lines
```
diff --git a/CHANGELOG.md b/CHANGELOG.md
new file mode 100644
index 00000000..0feaec93
--- /dev/null
+++ b/CHANGELOG.md
@@ -0,0 +1,5 @@
+# Changelog^M
+^M
+- Slimmer Dockerfile settings.^M
+^M
+# [Unreleased]^M
```

rmharrison replied:
`RELEASENOTES.md` should be renamed `CHANGELOG.md`

rmharrison reviewed (changes requested):
On second glance, a few (minor )questions emerge.

jmfadeley commented:
@rmharrison looking at these concerns now. They're slightly out of scope for what I intended but since I'm here might as well.

jmfadeley committed:
Addition of the create and delete db scripts and some new yarn
inclusions.

jmfadeley commented:
@rmharrison Pushed an update that includes the `create_db` and `delete_db` scripts. You will need to run `yarn` in order to install a couple of new packages, then run `yarn test` to test the results. For the `Dockerfile`, you want it to run the whole gauntlet like we do on the other services?

rmharrison committed:
Microservice consistency, PR #67

- Update to use `yarn` instead of `npm`
- Run `dos2unix` on README.md
- Update `package.json` scripts
- linted

rmharrison commented:
@jmfadeley Thanks for resolving comments. Approved.
